runtime/syntax/man.vim

```diff
@@ -18,6 +18,11 @@ highlight default link manOptionDesc     Constant
 highlight default link manReference      PreProc
 highlight default link manSubHeading     Function
 
+if &filetype != 'man'
+  " Loaded by the location list TOC
+  finish
```

Comment:
Is it *that* complicated, though?  It's not really doing anything specific for the qf window.  It's stopping it before executing script that's specific to the `man` plugin.  It actually makes the syntax loadable by other syntaxes.

Re: highlight in TOC, I'm a pretty strong proponent of notion that "color is information".  In Vim's help, the only defined sections are lines preceded by a `===` line.  Meaningful subsections either have all caps, or end with `~`.  There's some checks for the lines ending in `~` because it's sometimes used for table headings or code examples.  So, there's going to be false positives.  I felt that the highlighting made it more clear as to *why* a line is in the TOC in this case.